pkg/controllers/vdb/onlineupgrade_reconciler.go

```diff
@@ -923,10 +923,11 @@ func (r *OnlineUpgradeReconciler) getNewSandboxName(preferredName string) (strin
 		sbNames[r.VDB.Spec.Sandboxes[i].Name] = true
 	}
 
-	// To make this easier to test, we will favor the preferredName as the
+	// To make this easier to test, we will favor the annotation's value as the
 	// sandbox name. If that's available that's our name.
-	if _, found := sbNames[preferredName]; !found {
-		return preferredName, nil
+	sbName := vmeta.GetOnlineUpgradePreferredSandboxName(r.VDB.Annotations)
```

Comment:
No we don't because it is explicitly set by the user. We only clean annotations that are internally set. It is the user responsibility to manipulate it. I will add a comment to make that clear.

Reply:
Why does the user need this annotation? It will cause the second upgrade to fail if this annotation is still there. Operator should automatically set this sandbox name for the user.

Reply:
This annotation will not be advertised and will only be used for testing. In general if a user manually sets an annotation, the operator should not touch it to avoid confusion about why the annotation is not there anymore.
I do see where you are coming from and I can remove it at the end of online upgrade.